src/explorer/webview.ts

```diff
@@ -1058,29 +1093,29 @@ class ExplorerWebview {
      * Callback when the user clicks the Collapse all button
      * @param event event fired when user clicked Collapse all button
      */
-    private onCollapseAllClicked() {
+    private onCollapseAllClicked(): void {
         this.toggleTreeElements(ToggleState.expanded, ToggleState.collapsed);
     }
 
     /**
      * Callback when the user clicks the Expand all button
      * @param event event fired when user clicked Expand all button
      */
-    private onExpandAllClicked() {
+    private onExpandAllClicked(): void {
         this.toggleTreeElements(ToggleState.collapsed, ToggleState.expanded);
     }
 
     /**
      * Callback when user clicks on the Fix tree
      * @param event event fired when user clicked the fix tree
      */
-    private onFixClicked(event: MouseEvent) {
-        let ele = event.srcElement as HTMLElement;
+    private onFixClicked(event: MouseEvent): void {
+        let ele: HTMLElement | null = <HTMLElement>event.srcElement;
```

Comment:
Reason for the switch away from `as` to `<>` in the several of the lines here and below?

Reply:
I really want all the "Casts" to go away. The idea (which I have implemented) is a "type safe way" to make sure the cast is of the correct type.

---
In reply to: [397709904](https://github.com/microsoft/sarif-vscode-extension/pull/211#discussion_r397709904) [](ancestors = 397709904)